src/build/__main__.py

```diff
@@ -126,6 +119,56 @@ def build_package(srcdir, outdir, distributions, config_settings=None, isolation
         _error(str(e))
 
 
+def build_package(srcdir, outdir, distributions, config_settings=None, isolation=True, skip_dependency_check=False):
+    # type: (str, str, List[str], Optional[ConfigSettingsType], bool, bool) -> None
+    """
+    Run the build process.
+
+    :param srcdir: Source directory
+    :param outdir: Output directory
+    :param distribution: Distribution to build (sdist or wheel)
+    :param config_settings: Configuration settings to be passed to the backend
+    :param isolation: Isolate the build in a separate environment
+    :param skip_dependency_check: Do not perform the dependency check
+    """
+    with _handle_build_error():
+        builder = ProjectBuilder(srcdir)
+        for distribution in distributions:
+            _build(isolation, builder, outdir, distribution, config_settings, skip_dependency_check)
+
+
+def build_package_via_sdist(srcdir, outdir, distributions, config_settings=None, isolation=True, skip_dependency_check=False):
+    # type: (str, str, List[str], Optional[ConfigSettingsType], bool, bool) -> None
+    """
+    Build a sdist and then the specified distributions from it.
+
+    :param srcdir: Source directory
+    :param outdir: Output directory
+    :param distribution: Distribution to build (only wheel)
+    :param config_settings: Configuration settings to be passed to the backend
+    :param isolation: Isolate the build in a separate environment
+    :param skip_dependency_check: Do not perform the dependency check
+    """
+    if 'sdist' in distributions:
+        raise ValueError('Only binary distributions are allowed but sdist was specified')
+
+    with _handle_build_error():
+        builder = ProjectBuilder(srcdir)
+        sdist = _build(isolation, builder, outdir, 'sdist', config_settings, skip_dependency_check)
+
+        # extract sdist
+        sdist_name = os.path.basename(sdist)
+        sdist_out = tempfile.mkdtemp(dir=outdir, prefix='build-via-sdist-')
+        with tarfile.open(sdist) as t:
+            t.extractall(sdist_out)
+            builder = ProjectBuilder(os.path.join(sdist_out, sdist_name.rstrip('.tar.gz')))  # noqa: B005
```

Comment:
`str.rstrip` will mangle filenames with local version identifiers, slice the string instead: `sdist_name[: -len('.tar.gz')]`.

Reply:
This is why B005 exists, and [exactly why](https://www.python.org/dev/peps/pep-0616/#rationale) they added `.remove` suffix in Python 3.9.

Reply:
Right :facepalm:! It seems my memory failed me as I rushed through  this PR.